api/include/opentelemetry/trace/trace_state.h

```diff
@@ -107,11 +107,35 @@ class TraceState
     }
   };
 
+  // Creates an empty TraceState.
+  TraceState() noexcept : entries_(new Entry[kMaxKeyValuePairs]), num_entries_(0) {}
+
+  TraceState(const TraceState &other) noexcept
+      : entries_(new Entry[kMaxKeyValuePairs]), num_entries_(other.num_entries_)
+  {
+    for (size_t i = 0; i < num_entries_; i++)
+    {
+      entries_.get()[i] = other.entries_.get()[i];
+    }
+  }
+
+  TraceState operator=(const TraceState &other) noexcept
```

Comment:
`TraceState` is supposed to be immutable, so assignment operator should be deleted by default. We may need to explicitly add the assignment operator for `SpanContext` which does assignment for rest of it's members ( and not TraceState). As per specs:
``Please note, since SpanContext is immutable, it is not possible to update SpanContext with a new TraceState. ``